src/auro-combobox.js

```diff
@@ -66,6 +66,11 @@ export class AuroCombobox extends LitElement {
      */
     this.runtimeUtils = new AuroLibraryRuntimeUtils();
 
+    /**
+     * @private
+     */
+    this.observer = new MutationObserver(() => this.handleDeepSlotChanges());
```

Comment:
Does this also mean though that we would be rebuilding the menu option on _any_ change to the menu options, e.g. making a selection which would add the checkmark icon?

Reply:
There is an option that is false by default that can check for changes to attributes. We should be safe from triggering a change on toggling the `nocheckmark` attribute. https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe#attributes

Reply:
I'm seeing the mutation observer fire when I make a selection from the menu but otherwise do not modify the menu. This may not be a problem but I want to make sure we fully understand the implications of this change.

Let's get some pair programming time on this Monday.

Reply:
After walking through this with @rmenner the mutation observer logic makes sense and I agree it is doing the right thing.